include/component/switch.hh

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include "core/util.hh"
+#include "core/parameter.hh"
+
+// instead of using cv.hh, we directly use juce::CachedValue<bool>
+
+namespace box{
+struct Switch{
+    juce::CachedValue<bool> &value_;
+    uint8_t x_, y_;
+    Color colour_;
+    std::string name_;
+
+    Switch(juce::CachedValue<bool> &value, uint8_t x, uint8_t y, std::string name="")
+    : value_{value}, x_{x}, y_{y}, name_{name}, colour_{RED}
+    {}
+
+    virtual void Render(Interface &interface) 
+    {
+        colour_ = value? RED : WHITE;
```

Comment:
`color_` should be constant. you can create a local variable `color` depending on the state of `value_`, and then `DrawText()` with that.

Also, is `value` (no trailing underscore) a typo? how does this compile?